internal/bridge/client.go

```diff
@@ -45,6 +46,172 @@ type Client struct {
+}
+
+type ClusterUpgradeApiResource struct {
+	ClusterID       string                      `json:"cluster_id,omitempty"`
+	Operations      []*v1beta1.UpgradeOperation `json:"operations,omitempty"`
+	Team            string                      `json:"team_id,omitempty"`
+	ResponsePayload v1beta1.SchemalessObject    `json:""`
+}
+
+func (c *ClusterUpgradeApiResource) AddDataToClusterStatus(cluster *v1beta1.CrunchyBridgeCluster) {
+	cluster.Status.OngoingUpgrade = c.Operations
+	cluster.Status.Responses.Upgrade = c.ResponsePayload
+}
+
+type ClusterUpgradeOperationApiResource struct {
+	Flavor       string `json:"flavor,omitempty"`
+	StartingFrom string `json:"starting_from,omitempty"`
+	State        string `json:"state,omitempty"`
+}
+
+// BRIDGE API REQUEST PAYLOADS
+
+// PatchClustersRequestPayload is used for updating various properties of an existing cluster.
+type PatchClustersRequestPayload struct {
+	ClusterGroup string `json:"cluster_group_id,omitempty"`
+	// DashboardSettings      *ClusterDashboardSettings `json:"dashboard_settings,omitempty"`
+	// TODO: (dsessler7) Find docs for DashboardSettings and create appropriate struct
+	Environment            string `json:"environment,omitempty"`
+	IsProtected            bool   `json:"is_protected,omitempty"`
+	MaintenanceWindowStart int64  `json:"maintenance_window_start,omitempty"`
+	Name                   string `json:"name,omitempty"`
+}
+
+// PostClustersRequestPayload is used for creating a new cluster.
+type PostClustersRequestPayload struct {
+	Name            string             `json:"name"`
+	Plan            string             `json:"plan_id"`
+	Team            string             `json:"team_id"`
+	ClusterGroup    string             `json:"cluster_group_id,omitempty"`
+	Environment     string             `json:"environment,omitempty"`
+	IsHA            bool               `json:"is_ha,omitempty"`
+	Keychain        string             `json:"keychain_id,omitempty"`
+	Network         string             `json:"network_id,omitempty"`
+	PostgresVersion intstr.IntOrString `json:"postgres_version_id,omitempty"`
+	Provider        string             `json:"provider_id,omitempty"`
+	Region          string             `json:"region_id,omitempty"`
+	Storage         int64              `json:"storage,omitempty"`
+}
+
+// PostClustersUpgradeRequestPayload is used for creating a new cluster upgrade which may include
+// changing its plan, upgrading its major version, or increasing its storage size.
+type PostClustersUpgradeRequestPayload struct {
+	Plan             string             `json:"plan_id,omitempty"`
+	PostgresVersion  intstr.IntOrString `json:"postgres_version_id,omitempty"`
+	UpgradeStartTime string             `json:"starting_from,omitempty"`
+	Storage          int64              `json:"storage,omitempty"`
+}
+
+// PutClustersUpgradeRequestPayload is used for updating an ongoing or scheduled upgrade.
+type PutClustersUpgradeRequestPayload struct {
```

Comment:
Are we going to be doing this?

Reply:
Not sure. Maybe at some point. I guess this is just there as a placeholder.